src/components/Firestore/dialogs/AddDocumentDialog.test.tsx

```diff
@@ -18,70 +18,74 @@ import { fireEvent } from '@testing-library/react';
 import React from 'react';
 import { act } from 'react-dom/test-utils';
 
-import { renderDialog, waitForDialogsToClose } from '../../../test_utils';
 import {
-  fakeCollectionReference,
-  fakeDocumentReference,
-} from '../testing/models';
+  renderDialogWithFirestore,
+  waitForDialogsToClose,
+} from '../../../test_utils';
 import { AddDocumentDialog } from './AddDocumentDialog';
 
-const docRef = fakeDocumentReference({
-  id: 'random-identifier',
-});
-const collectionReference = fakeCollectionReference({
-  id: 'my-stuff',
-  path: 'users/bob/my-stuff',
-  doc: jest.fn(),
-});
-
-beforeEach(() => {
-  collectionReference.doc.mockReturnValue(docRef);
-});
-
 it('shows correct title', async () => {
-  const { getByText } = await renderDialog(
-    <AddDocumentDialog
-      open={true}
-      collectionRef={collectionReference}
-      onValue={() => {}}
-    />
-  );
+  const { getByText } = await renderDialogWithFirestore(async firestore => {
+    const collectionRef = firestore.collection('things');
+    return (
+      <AddDocumentDialog
+        open={true}
+        collectionRef={collectionRef}
+        onValue={() => {}}
+      />
+    );
+  });
 
   expect(getByText(/Add a document/)).not.toBeNull();
 });
 
 it('shows the (disabled) creation path', async () => {
-  const { getByLabelText } = await renderDialog(
-    <AddDocumentDialog
-      open={true}
-      collectionRef={collectionReference}
-      onValue={() => {}}
-    />
+  const { getByLabelText } = await renderDialogWithFirestore(
+    async firestore => {
+      const collectionRef = firestore.collection('things');
```

Comment:
Let's use a nested collection here, since that's what the original testing is for and that is more interesting.

Reply:
Ack, updated